pulsar-client-api/src/main/java/org/apache/pulsar/client/api/schema/GenericRecord.java

```diff
@@ -61,4 +64,29 @@ default Object getField(Field field) {
      */
     Object getField(String fieldName);
 
+    /**
+     * Return the schema tyoe.
+     *
+     * @return the schema type
+     * @throws UnsupportedOperationException if this feature is not implemented
+     * @see SchemaType#AVRO
+     * @see SchemaType#PROTOBUF_NATIVE
+     * @see SchemaType#JSON
+     */
+    default SchemaType getSchemaType() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Return the internal native representation of the Record,
+     * like a AVRO GenericRecord.
+     * You have to pass the type you would like to obtain.
+     * This method will return null if such type is not supported.
+     *
+     * @return the internal representation of the record, or null if the requested information is not available.
+     */
+    default <T> T getNativeRecord(Class<T> clazz) {
+        return null;
```

Comment:
I am confused why do you throw UnsupportedOperationException for `getSchemaType` and return `null` for this method.

Reply:
because we do not have a good "default value" for `getSchemaType`

but for this method the expected behaviour is that you get a `null` value if you are asking for an unsupported interface, so returning `null` here is the expected behaviour.

Reply:
Why not throw UnsupportedOperationException for `getNativeRecord`? It is a much clear behavior than returning `null`.

Reply:
I believe that `UnsupportedOperationException` does not work well, 
because code that calls that method will have to catch it, and catching `UnsupportedOperationException` smells.

I thinking about generic code that handles multiple different types of records.